### PR TITLE
Feat/derivation test cache

### PR DESCRIPTION
**Pull request by matiu**
* Cache key derivation test, using client's user-agent.
* refactor angular-bwc's config to make it stateless.

### CODE REVIEW

**Comment by matiu** [comment]
updated. Test fixed.

File: src/js/services/profileService.js [hunk 1]
```diff
@@ -62,12 +62,11 @@ angular.module('copayApp.services')
       });
     };
 
-    root.setBaseURL = function(walletId) {
+    root._getBaseURL = function(walletId) {
```
Comment (JDonadio) [inline src/js/services/profileService.js]: Should not be "root" if it's an internal function

File: src/js/services/storageService.js [hunk 2]
```diff
@@ -113,11 +113,14 @@ angular.module('copayApp.services')
     };
 
     root.storeProfile = function(profile, cb) {
+console.log('[storageService.js.115:storeProfile:]',profile); //TODO
```
Comment (JDonadio) [inline src/js/services/storageService.js]: Remove console.log

**Comment by matiu** [comment]
updated. Thanks for the comments

**Comment by JDonadio** [comment]
ACK

**Pull request merged**